pkg/didcomm/packager/packager.go

```diff
@@ -117,30 +118,40 @@ func (bp *Packager) PackMessage(messageEnvelope *transport.Envelope) ([]byte, er
 	return marshalledEnvelope, nil
 }
 
-//nolint:funlen,gocyclo
+//nolint:funlen,gocyclo,gocognit
 func (bp *Packager) prepareSenderAndRecipientKeys(cty string, envelope *transport.Envelope) ([]byte, [][]byte, error) {
 	var recipients [][]byte
 
 	isLegacy := isMediaTypeForLegacyPacker(cty)
 
-	for i, receiverKey := range envelope.ToKeys {
+	for i, receiverKeyID := range envelope.ToKeys {
 		switch {
-		case strings.HasPrefix(receiverKey, "did:key"):
-			marshalledKey, err := addDIDKeyToRecipients(i, receiverKey, isLegacy)
+		case strings.HasPrefix(receiverKeyID, "did:key"):
+			marshalledKey, err := addDIDKeyToRecipients(i, receiverKeyID, isLegacy)
 			if err != nil {
 				return nil, nil, err
 			}
 
 			recipients = append(recipients, marshalledKey)
-		case strings.Index(receiverKey, "#") > 0:
-			marshalledKey, err := bp.resolveKeyAgreementFromDIDDoc(receiverKey)
+		case strings.Index(receiverKeyID, "#") > 0:
```

Comment:
this condition is not meant for legacy packing

Reply:
and yet, some did docs in legacy interop scenarios _do_ use DID url key references - and the new handling, that assumes this only happens in non-legacy scenarios, broke interop tests that previously worked.